### ISSUE TITLE
Material-UI 5.0 support

### ISSUE THREAD

**Issue by tbowmo** [iamhosseindhv/notistack#412]
## Expected Behavior
Project should be able to compile

## Current Behavior
notistack imports makeStyle and friends from @material-ui/core/styles, which doesn't seem to export these functions anymore, and thus we need to update the imports to import from @material-ui/styles instead.

## Steps to Reproduce
Upgrade an existing project to material-ui 5, or create a test project.

## Context
I'm trying to upgrade to material-ui 5.0 (it's in beta at the moment), to see if that solves some of my issues with an autocomplete form, but then it breaks functionality with notistack.

## Your Environment

| Tech         | Version |
|--------------|---------|
| Notistack | v1.0.9  |
| React        |  17.02    |
| Material-UI | 5.0.0-beta |
| etc.         |         |

**Comment by evantrimboli** [comment]
I've seen some posts indicating that this `@next` versions is supposed to solve this issue: https://www.npmjs.com/package/notistack/v/1.0.6-next.1

It didn't work for me and I've got the same issue, but hopefully YMMV.

**Comment by Brian-McBride** [comment]
No, it seems that the current build still imports `makeStyle` and `withStyles`.

The styling here really needs to be redone to match Emotion styled.

**Comment by MaddyLeadbetter** [comment]
I'm using MUI v5.0.0-beta.1 and notistack v1.0.6-next.1, and also getting the withStyles issues.  Are there plans to release a new version that supports v5 beta?

**Comment by iamhosseindhv** [comment]
Please have look through PR #387 and read [this comment](https://github.com/iamhosseindhv/notistack/pull/387#issuecomment-881417683) to see why it was closed.

TLDR: We need to implement the solution suggested on the comment above. I have to implement this on my free time so PRs are welcome to save time and make notistack compatible as soon as possible.

**Comment by mwhitney-butter** [comment]
any eta on material-ui v5 support?

**Comment by FlexibleHha** [comment]
notistack5": "^1.0.1",  Could not load './dist/notistack.esm.js' from module 'notistack5' found in package.json#module

**Comment by skychavda** [comment]
I am currently working on my project and notistack is giving error of missing `makeStyles`

Tech | Version
--- | --- 
Notistack | v1.0.10
React | 17.0.2
MaterialUI | 5.0.0-alpha.38

![image](https://user-images.githubusercontent.com/36260957/129325649-1506be68-4651-4e7a-9d70-1f88be0cb927.png)

**Comment by joemaffei** [comment]
I just posted a PR for upgrading to v5. https://github.com/iamhosseindhv/notistack/pull/418

**Comment by Brian-McBride** [comment]
Maybe this project isn't as actively maintained?

**Comment by iamhosseindhv** [comment]
Thanks to @joemaffei, you can now download a MUI v5 beta compatible version using  `npm i notistack@next`. Please open a new issue if you're having any issues.
